kolibri/plugins/management/assets/src/vue/manage-class-page/class-delete-modal.vue

```diff
@@ -68,14 +62,10 @@
         required: true,
       },
     },
-    data() {
-      return {
-        error_message: '',
-      };
-    },
     methods: {
-      submit() {
-        // need delete logic here.
+      classDelete() {
+        this.deleteClass(this.classid);
+        this.$emit('close');
```

Comment:
> I thought you oppose the idea to have action directly control the view via promise.

I do, but I also agree with you that there might be legitimate use-cases for it to help prevent vuex state bloat. At the very least, it's good to try different patterns until we find one that seems to work well.

> I don't see how this solution solves mutation bloating

It helps because most state gets pushed to the local `data` in the view. Far fewer vuex state gets added, mostly relating to whether or not modals are visible.

> when `modalShown` is set to null or 0, all modals hide
> when `modalShown` is set to `classNameEdit` string, only class-name-edit modal shown.

Yeah, that's roughly what I was thinking - as long as those strings are defined in a constants file.